codegen/src/test/function.rs

```diff
@@ -237,11 +237,14 @@ mod generate_tests {
         let expected = expected.to_string();
         if &actual != &expected {
             let mut counter = 0;
-            let iter = actual
-                .chars()
-                .zip(expected.chars())
-                .inspect(|_| counter += 1)
-                .skip_while(|(a, e)| *a == *e);
+            let iter = actual.chars().zip(expected.chars()).skip_while(|(a, e)| {
```

Comment:
I think this is more difficult to read than the original, because I have to think about the returns and the full conditional. Was this a suggestion by clippy?

Reply:
No.  `counter` has the wrong value.  It is 50 characters too many when I turned on the 50-character diff print-out.

That's because the `inspect` clause is run even for the diff print-out.  `counter += 1` must be moved inside `skip_while`.